components/browser/tabstray/src/main/java/mozilla/components/browser/tabstray/TabsAdapter.kt

```diff
@@ -24,16 +25,18 @@ typealias ViewHolderProvider = (ViewGroup, BrowserTabsTray) -> TabViewHolder
  */
 @Suppress("TooManyFunctions")
 open class TabsAdapter(
-    delegate: Observable<TabsTray.Observer> = ObserverRegistry(),
+    thumbnailLoader: ImageLoader? = null,
     private val viewHolderProvider: ViewHolderProvider = { parent, tabsTray ->
         DefaultTabViewHolder(
                 LayoutInflater.from(parent.context).inflate(
                         R.layout.mozac_browser_tabstray_item,
                         parent,
                         false),
-                tabsTray
+                tabsTray,
+                thumbnailLoader
         )
-    }
+    },
+    delegate: Observable<TabsTray.Observer> = ObserverRegistry()
```

Comment:
would putting this parameter after the closure prevent us from using the trailing closures when initializing the `TabsAdapter`?

Reply:
@boek I explicitly wanted to avoid that since we've had issues (https://github.com/mozilla-mobile/fenix/pull/10495) where the scope of the trailing closure did not make it obvious it was being invoked during the `onCreateViewHolder` call of the adapter and that could cause crashes depending on what work was done in that closure.

I thought it was better to have an explicit named param in that case to avoid regressions.